components/raftstore/src/store/entry_storage.rs

```diff
@@ -530,6 +551,63 @@ pub fn init_applied_term<ER: RaftEngine>(
     }
 }
 
+/// When a peer(follower) receives a TransferLeaderMsg, it enters the
+/// CacheWarmupState. When the peer becomes leader or it doesn't
+/// become leader before a deadline, it exits the state.
+#[derive(Clone, Debug)]
+pub struct CacheWarmupState {
+    range: (u64, u64),
+    is_task_timeout: bool,
+    started_at: Instant,
+}
+
+impl CacheWarmupState {
+    pub fn new() -> Self {
+        CacheWarmupState::new_with_range(INVALID_INDEX, INVALID_INDEX)
+    }
+
+    pub fn new_with_range(low: u64, high: u64) -> Self {
+        CacheWarmupState {
+            range: (low, high),
+            is_task_timeout: false,
+            started_at: Instant::now(),
+        }
+    }
+
+    pub fn range(&self) -> (u64, u64) {
+        self.range
+    }
+
+    /// How long has it been in this state.
+    pub fn elapsed(&self) -> Duration {
+        self.started_at.saturating_elapsed()
+    }
+
+    /// Whenther the warmup task is timeout.
+    pub fn is_task_timeout(&self) -> bool {
+        self.is_task_timeout
+    }
+
+    pub fn mark_task_timeout(&mut self) {
+        WARM_UP_ENTRY_CACHE_COUNTER.timeout.inc();
+        self.is_task_timeout = true;
+    }
+
+    /// Whether the state is stale.
+    ///
+    /// Exit the state when the state is stale.
+    pub fn is_stale(&self) -> bool {
```

Comment:
unify it with `is_task_timeout` like
```rust
pub fn check_timeout(&mut self, duration: Duration) -> bool {
    if self.started_at.elapsed() > duration {
        self.timeout = true;
    }
    self.timeout
}
```

Reply:
I changed the `mark_task_timeout` to `check_task_timeout` and I think it is reasonable.For `is_stale`, I'm not sure if it should be also changed to `check_stale`, and I think it brings little benefit 🤔.

PTAL again ❤️